geode-core/src/main/java/org/apache/geode/cache/client/internal/QueueManagerImpl.java

```diff
@@ -802,10 +803,13 @@ private QueueConnectionImpl createNewPrimary(Set<ServerLocation> excludedServers
       excludedServers.addAll(servers);
     }
 
+    return primary;
```

Comment:
My only concern is that after this change, `createNewPrimary()` method doesn't send `readyForEvent` to the server anymore. This method has two usages: `recoveryPrimary` and `initializeConnections`, `readyForEvent` will be sent by the first usage. As long as we know for sure it has no impact on the 2nd usage. Then this fix should be fine.

Reply:
@mhansonp were you able to find an answer to Jinmei's question?

Reply:
@pivotal-amurmann Yes, I did. There are two calls and I have made sure that we are calling `markQueueAsReadyForEvents` after each one. So, it is covered.

Reply:
I am still working out the unit tests.